### DISCUSSION TITLE
How does AIR obtain an IP on the Android system?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#2633] [Q&A]
Under Windows system, the local IP can be obtained through the following code, but not on Android.
May I ask if my method is incorrect?
thank you!

var networkInfo:NetworkInfo = NetworkInfo.networkInfo; 
        var interfaces:Vector.<NetworkInterface> = networkInfo.findInterfaces(); 
         
        if( interfaces != null ) 
        { 
            pp_txt.appendText( "Interface count: " + interfaces.length ); 
            for each ( var interfaceObj:NetworkInterface in interfaces ) 
            { 
                pp_txt.appendText( "\nname: "             + interfaceObj.name ); 
                pp_txt.appendText( "display name: "     + interfaceObj.displayName ); 
                pp_txt.appendText( "mtu: "                 + interfaceObj.mtu ); 
                pp_txt.appendText( "active?: "             + interfaceObj.active ); 
                pp_txt.appendText( "parent interface: " + interfaceObj.parent ); 
                pp_txt.appendText( "hardware address: " + interfaceObj.hardwareAddress ); 
                if( interfaceObj.subInterfaces != null ) 
                { 
                    pp_txt.appendText( "# subinterfaces: " + interfaceObj.subInterfaces.length ); 
                } 
                pp_txt.appendText("# addresses: "     + interfaceObj.addresses.length ); 
                for each ( var address:InterfaceAddress in interfaceObj.addresses ) 
                { 
                    pp_txt.appendText( "  type: "           + address.ipVersion ); 
                    pp_txt.appendText( "  address: "         + address.address ); 
                    pp_txt.appendText( "  broadcast: "         + address.broadcast ); 
                    pp_txt.appendText( "  prefix length: "     + address.prefixLength ); 
                } 
            }             
        }

**Comment by teotigraphix** [comment]
I don't have the answer but I have used UDP/OSC successfully on an Android device with AIR.

That said, it was years ago so maybe, Android has permissions for local network discovery?

One other thing, recently Harman took out the injected INTERNET permission. I am guessing, that if you don't add that permission in your manifest, Android is blocking your ports. (permission denied might be in logcat?)

**Comment by Jason0247** [comment]
Sure enough, my permissions were not enabled. Unlike iOS, I ignored the permissions issue on Android. Thank you~ @teotigraphix
